api/services/AssetManagement.js

```diff
@@ -1,68 +1,51 @@
-const aws = require('aws-sdk')
-const crypto = require('crypto')
-const gm = require('gm')
-const mime = require('mime')
-const path = require('path')
-const Promise = require('bluebird')
-const request = require('request')
-
+import crypto from 'crypto'
+import path from 'path'
+import Promise from 'bluebird'
+import request from 'request'
+import sharp from 'sharp'
+import { createS3StorageStream } from './Uploader/storage'
+
+// this defends against filenames that are blank or have troublesome characters
```

Comment:
@levity i've used this https://www.npmjs.com/package/sanitize-filename in the past.  could be useful...